--- /src/lang/perl/modules/libnet-1.12/src/Net/Time.pm	Fri Aug 17 10:05:27 2001
+++ Net/Time.pm	Fri Aug  2 01:17:21 2002
@@ -61,7 +61,7 @@
  my $offset = 0 | 0;
 
  return undef
-	unless $s->recv($buf, length(pack("N",0)));
+	unless defined $s->recv($buf, length(pack("N",0)));
 
  # unpack, we | 0 to ensure we have an unsigned
  my $time = (unpack("N",$buf))[0] | 0;
@@ -87,7 +87,7 @@
  my $s = _socket('daytime',13,@_) || return undef;
  my $buf = '';
 
- $s->recv($buf, 1024) ? $buf
+ defined $s->recv($buf, 1024) ? $buf
     	              : undef;
 }
 
